plugin/src/main/java/org/opensearch/sql/plugin/catalog/CatalogServiceImpl.java

```diff
@@ -113,7 +118,11 @@ private StorageEngine createStorageEngine(CatalogMetadata catalog) throws URISyn
     ConnectorType connector = catalog.getConnector();
     switch (connector) {
       case PROMETHEUS:
-        storageEngine = null;
+        PrometheusClient
+            prometheusClient =
+            new PrometheusClientImpl(new OkHttpClient(),
+                new URI(catalog.getUri()));
```

Comment:
Is this client created per query request?

Reply:
Only once during the start of the server.